message.go

```diff
@@ -294,13 +300,44 @@ func doParsing(mp *msgParser) (err error) {
 		}
 	}
 
-	bodyLength, err := mp.msg.Header.getIntNoLock(tagBodyLength)
+	var bodyLength int
+	bodyLength, err = mp.msg.Header.getIntNoLock(tagBodyLength)
 	if err != nil {
 		err = parseError{OrigError: err.Error()}
 	} else if length != bodyLength && !xmlDataMsg {
 		err = parseError{OrigError: fmt.Sprintf("Incorrect Message Length, expected %d, got %d", bodyLength, length)}
 	}
 
+	mp.msg.Header.rwLock.Unlock()
+	mp.msg.Body.rwLock.Unlock()
+	mp.msg.Trailer.rwLock.Unlock()
+	locked = false
```

Comment:
This is an antipattern, if you need to reduce the locked surface use a dedicated function with a defer at the top as this function previously had